spec/support/elasticsearch_helpers.rb

```diff
@@ -0,0 +1,66 @@
+module ElasticsearchHelpers
+  def put_template_for_ingest_measurements(client)
+    client.indices.put_template(
+      name: 'ingest-measurements',
+      body: {
+        template: 'ingest-measurements-*',
+        order: 0,
+        settings: {
+          number_of_shards: 1,
+          number_of_replicas: 0
+        },
+        mappings: {
+          _default_: {
+            _all: {
+              norms: false
+            },
+            _meta: {
+              version: '1.0.0'
+            },
+            date_detection: false,
+            dynamic_templates: [
+              {
+                strings_as_keyword: {
+                  mapping: {
+                    ignore_above: 1024,
+                    type: 'keyword'
+                  },
+                  match_mapping_type: 'string'
+                }
+              }
+            ],
+            properties: {
+              :@timestamp => {
+                type: 'date'
+              },
+              service_uploaded: {
+                type: 'date'
+              },
+              :'ingest.location' => {
+                type: 'geo_point',
+                ignore_malformed: true
+              },
+              pms_std01_0: {
+                type: 'float'
+              }
+            }
+          }
+        }
+      }
+    )
+  end
+
+  def delete_ingest_measurements_indicies(client)
+    client.indices.delete(index: 'ingest-measurements-*')
+  end
+
+  def add_index_to_ingest_measurements(client, body)
+    client.index(
+      index: 'ingest-measurements-2018-12-01',
+      type: 'measurement',
+      body: body.merge('@timestamp': '2018-12-01T12:34:56Z')
+    )
+    # Need to have some period to get result
```

Comment:
Interesting. I get the impression there's a method we can use on client.index to force the result

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#index-wait-for-active-shards

Maybe this?